### ISSUE TITLE
Producing potentially unsafe @keyframes rules

### ISSUE THREAD

**Issue by AlanGreene** [ben-eb/postcss-merge-rules#23]
Some browsers, e.g. Safari, do not support cascading in `@keyframes` rules.

Using cssnano, example input:
```css
@keyframes foo {
  0% {
    visibility: visible;
    transform: scale3d(.85, .85, .85);
    opacity: 0;
  }
  100% {
    visibility: visible;
    opacity: 1;
  }
}
```

produces this potentially unsafe output:
```css
@keyframes foo {
  0% {
    transform: scale3d(.85, .85, .85);
    opacity: 0;
  }
  0%,100% {
    visibility: visible;
  }
  100% {
    opacity: 1;
  }
}
```

Since Safari does not support cascading in these rules this will not work as the `opacity` and `transform` properties from the 0% frame will be ignored.

Disabling mergeRules produces valid output that works in all modern browsers supporting `@keyframes`, including Safari.

**Comment by ben-eb** [comment]
Should we just skip `@keyframes` merging altogether? My understanding of the specification is that rather than merging this example, the *last* value applies;

```css
@keyframes demo {
  50% {
    color: red;
  }
  50% {
    opacity: 0.5;
  }
}
```

So the browser will actually drop the first declaration entirely, and renders the CSS as if it were written like this:

```css
@keyframes demo {
  50% {
    opacity: 0.5;
  }
}
```

In this case we probably need a new algorithm. What do you think?

/cc @TrySound

**Comment by AlanGreene** [comment]
I'd definitely give my :+1: for skipping `@keyframes` merging altogether as it would result in valid output for all browsers. Perhaps once the spec stabilizes and browsers are updated to support cascading this could be added back again.

**Comment by TrySound** [comment]
@ben-eb Seems like we shouldn't merge keyframes as said in spec

> The keyframes rule that is used by an animation will be the last one encountered in sorted rules order that matches the name of the animation specified by the ‘animation-name’ property. keyframes rules do not cascade; therefore, an animation will never derive keyframes from more than one keyframes rule.

**Comment by TrySound** [comment]
However chrome use the last one not in order.

**Comment by ben-eb** [comment]
Released in `2.0.8`, thanks for the report! 👍

**Comment by AlanGreene** [comment]
thanks guys, looks good
